### PR TITLE
Get resources directly from redis without conf files

### LINKED ISSUE #45: Gateway takes a long time to become ready after nginx reload
Right now, we are using the `filemgmt` module to update conf files and reload nginx when new APIs are created. When the number of APIs increases (ie > 10000), `nginx reload` takes a long time to get set up (>2 min). We can instead go get the data straight from redis to avoid this problem.

### PR DESCRIPTION
**Pull request by alexsong93**
Fixes #45. Also refactored the management interface. @mhamann @DavidMGreen PTAL

### CODE REVIEW

File: api-gateway-config/scripts/lua/lib/redis.lua [hunk 1]
```diff
@@ -411,39 +382,22 @@ function _M.subscribe(redisSubClient, redisGetClient)
...
 end
 
 --- Get gateway sync status
 function _M.healthCheck()
-  if getSyncStatus() == true then
-    request.success(503, "Status: Gateway syncing.")
-  else
-    request.success(200, "Status: Gateway ready.")
-  end
+  request.success(200,  "Status: Gateway ready.")
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/lib/redis.lua]: Is there any case where the gateway might not be ready? (e.g. unable to connect to redis)

File: api-gateway-config/scripts/lua/routing.lua [hunk 2]
```diff
@@ -22,27 +22,37 @@
...
+function processCall()
+  -- Get resource object from redis
+  local red = redis.init(REDIS_HOST, REDIS_PORT, REDIS_PASS, 10000)
+  local redisKey = utils.concatStrings({"resources:", ngx.var.tenant, ":", ngx.var.gatewayPath})
+  local obj = redis.getResource(red, redisKey, "resources")
+  -- Check for path parameters
+  if obj == nil then
+    obj = checkForPathParams(red)
+    if obj == nil then
+      return request.err(404, 'API doesn\'t exist.')
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/routing.lua]: Let's change this to `Not found` just to make things nicely generic.

File: api-gateway-config/scripts/lua/lib/redis.lua [hunk 3]
```diff
@@ -411,39 +382,22 @@ function _M.subscribe(redisSubClient, redisGetClient)
       local redisKey = utils.concatStrings({resourcePrefix, ":", tenant, ":", gatewayPath})
       -- Don't allow single quotes in the gateway path
       if string.match(gatewayPath, "'") then
-        logger.debug(utils.concatStrings({"Redis key \"", redisKey, "\" contains illegal character \"'\"."}))
+        logger.info(utils.concatStrings({"Redis key \"", redisKey, "\" contains illegal character \"'\"."}))
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/lib/redis.lua]: This seems like it should be more of a warning / error to me. What happens if it contains an illegal character?

Comment (alexsong93) [inline api-gateway-config/scripts/lua/lib/redis.lua]: It notes that the redis key that was created has an illegal character and ignores it, but continues with the subscribe

File: api-gateway-config/scripts/lua/lib/redis.lua (hunk 1 above)
Comment (alexsong93) [inline api-gateway-config/scripts/lua/lib/redis.lua]: If the gateway is unable to connect to redis, it will retry a couple more times and then exit.

**Pull request merged**